### PR TITLE
Fix assert in AnalyzerDriver.GetOperationsToAnalyze

### LINKED ISSUE #26217: Assertion failed in AnalyzerDriver.GetOperationsToAnalyze
**Version Used**: 
image

**Steps to Reproduce**:

1. build Debug configuration of Roslyn.sln
2. `devenv /rootsuffix RoslynDev`
3. Create a console app with following code:

```C#
    class C
    {
        public C(int x)
        {
        }
    }

	class Program : C
	{
        int a;

        Program() : base(1) => this.a = 1;

        static void Main(string[] args)
        {
            new Program();
        }
    }
```

```
---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------
Expected operation with kind 'Invocation' to be the root operation with null 'Parent', but instead it has a non-null Parent with kind 'ExpressionStatement'



   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.GetOperationsToAnalyze(ImmutableArray`1 operationBlocks)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.TryExecuteDeclaringReferenceActions(SyntaxReference decl, Int32 declarationIndex, SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean shouldExecuteSyntaxNodeActions, Boolean shouldExecuteOperationActions, Boolean shouldExecuteCodeBlockActions, Boolean shouldExecuteOperationBlockActions, Boolean isInGeneratedCode, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.TryExecuteDeclaringReferenceActions(SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean isGeneratedCodeSymbol, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.TryProcessSymbolDeclared(SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.TryProcessEventCore(CompilationEvent e, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessEvent(CompilationEvent e, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.<ProcessCompilationEventsCoreAsync>d__70.MoveNext()

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessCompilationEventsCoreAsync(AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean prePopulatedEventQueue, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.<ProcessCompilationEventsAsync>d__69.MoveNext()

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ProcessCompilationEventsAsync(AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean prePopulatedEventQueue, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.<ExecutePrimaryAnalysisTaskAsync>d__46.MoveNext()

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.ExecutePrimaryAnalysisTaskAsync(AnalysisScope analysisScope, AnalysisState analysisStateOpt, Boolean usingPrePopulatedEventQueue, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.<AttachQueueAndProcessAllEventsAsync>d__44.MoveNext()

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.AttachQueueAndProcessAllEventsAsync(AsyncQueue`1 eventQueue, AnalysisScope analysisScope, AnalysisState analysisStateOpt, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.<ComputeAnalyzerDiagnosticsCoreAsync>d__64.MoveNext()

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.ComputeAnalyzerDiagnosticsCoreAsync(AnalyzerDriver driver, AsyncQueue`1 eventQueue, AnalysisScope analysisScope, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.<>c__DisplayClass57_2.<<ComputeAnalyzerDiagnosticsAsync>b__1>d.MoveNext()

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.<>c__DisplayClass57_2.<ComputeAnalyzerDiagnosticsAsync>b__1()

   at System.Threading.Tasks.Task`1.InnerInvoke()

   at System.Threading.Tasks.Task.Execute()

   at System.Threading.Tasks.Task.ExecutionContextCallback(Object obj)

   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)

   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)

   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()

   at System.Threading.ThreadPoolWorkQueue.Dispatch()

   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()


---------------------------
Abort   Retry   Ignore   
---------------------------

```

### LINKED ISSUE #43106: AnalyzerDriver.GetOperationsToAnalyze throws assert "Expected operation with kind 'Invocation' to be the root operation with null 'Parent', but instead it has a non-null Parent with kind 'ExpressionStatement'" on incomplete C# code
**Version Used**: 16.6 Preview 3

**Steps to Reproduce**:

1. Paste C#:

```C#
class B
{
    // Haven't typed { } on the next line yet
    public B() : this(1) 

    public B(int a) { } 
}
```

**Expected Behavior**: No asserts
**Actual Behavior**: Assert:

<details>
<summary>Expected operation with kind 'Invocation' to be the root operation with null 'Parent', but instead it has a non-null Parent with kind 'ExpressionStatement'</summary>
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.GetOperationsToAnalyze(ImmutableArray`1 operationBlocks)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.<>c__DisplayClass14_0.<TryExecuteDeclaringReferenceActions>g__getOperationsToAnalyzeWithStackGuard|4(ImmutableArray`1 operationBlocksToAnalyze)

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.<>c__DisplayClass14_0.<TryExecuteDeclaringReferenceActions>g__executeExecutableCodeActions|3()

   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver`1.TryExecuteDeclaringReferenceActions(SyntaxReference decl, Int32 declarationIndex, SymbolDeclaredCompilationEvent symbolEvent, AnalysisScope analysisScope, AnalysisState analysisStateOpt, GroupedAnalyzerActions additionalPerSymbolActions, Boolean shouldExecuteSyntaxNodeActions, Boolean shouldExecuteOperationActions, Boolean should......
</details>

### PR DESCRIPTION
**Pull request by mavasani**
Adjust the assert to handle the constructor initializer case.
Fixes #26217
Fixes #43106

### CODE REVIEW

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs [hunk 1]
```diff
@@ -2619,6 +2619,11 @@ private static bool IsEquivalentSymbol(ISymbol declaredSymbol, ISymbol otherSymb
                                 operationsToAnalyze.Add(operationBlock.Parent);
                                 break;
 
+                            case OperationKind.ExpressionStatement:
+                                // For constructor initializer, we generate an IInvocationOperation with an implicit IExpressionStatementOperation parent.
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs:2623]: >For constructor initializer, we generate an IInvocationOperation with an implicit IExpressionStatementOperation parent. [](start = 35, length = 119)

The assert looks less specific. Why don't we assert exactly what we expect, i.e.:
- the parent is implicit IExpressionStatementOperation
- the parent.Parent is IConstructorBodyOperation
- the parent is in initializer position within parent.Parent

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs (hunk 1 above)
Comment (mavasani) [reply src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs:2623]: Thanks, done.

**Comment by mavasani** [comment]
@dotnet/roslyn-compiler for additional review.

**Comment by mavasani** [comment]
Thank you!

**Pull request merged**